testutil/ibc_testing/generic_setup.go

```diff
@@ -52,7 +53,12 @@ func AddProvider[T e2eutil.ProviderApp](coordinator *ibctesting.Coordinator, t *
 	provider := ibctesting.NewTestChain(t, coordinator, appIniter, provChainID)
 	coordinator.Chains[provChainID] = provider
 
-	return provider, provider.App.(T)
+	providerToReturn, ok := provider.App.(T)
+	if !ok {
+		panic(fmt.Sprintf("provider app type returned from app initer does not match app type passed in as type param: %T, %T",
+			provider.App, *new(T)))
```

Comment:
See https://stackoverflow.com/questions/72610728/print-type-parameter-name-in-go